src/extractor/guidance/intersection_handler.cpp

```diff
@@ -51,6 +51,11 @@ TurnType::Enum IntersectionHandler::findBasicTurnType(const EdgeID via_edge,
     const auto &in_data = node_based_graph.GetEdgeData(via_edge);
     const auto &out_data = node_based_graph.GetEdgeData(road.eid);
 
+    // suppress instructions on edges that the can't control anyway
+    if (!in_data.startpoint && !out_data.startpoint)
```

Comment:
The problem I see here is that what we snap to and what we consider as connections in forms of a turn are two entirely different concepts to me. Ferries are special in terms of that they often share ways in OSM data but don't allow switching from one item to another in between.

If we want to specify that snapping to is equivalent to does not allow turning, then you would be right. But as long as we don't make a distinction here, using `startingpoint` would not capture the issue we see in ferries. 

The concept we need are ways that share segments but are distinct (right now we don't handle these correctly in all cases, since we can right now actually switch ferries in between). Ferries will need specific handling anyhow. The main question for me here is: do we really want to tie snapping to turn instructions?

/cc @daniel-j-h @danpat for additional oppinions